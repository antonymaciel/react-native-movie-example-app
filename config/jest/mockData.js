export const movie = {
    id: 0,
    year: '2019',
    imageUrl: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    title: 'Spiderman',
    vote_average: '9.1',
    urlbackdropImage: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg', 
    generes: 'Aventure, Drama, Terror',  
    vote_count: '1200',
    overview: 'Amazing movie',
    backdrop_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    genres: ['Adventure', 'Terror', 'Drama'],
    release_date: '2016-5-14'
}

export const review = {
    id: 0,
    author: 'Antony Maciel',
    url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
    content: 'Great Movie!, always has amazing content \n I love this movie'
}

export const movies = [
    {
        id: 1,
        poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'XMan',
        vote_average: 7.8,
        release_date: '2016-5-14'
    },
    {     
        id: 2,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Superman',
        vote_average: 0.9,
        release_date: '2015-5-9'
    },
    {
        id: 3,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Spiderman',
        vote_average: 8.0,
        release_date: '2005-5-27'
    }
]


export const reviews = [
    {
        id: 1,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!!!'
    },
    {
        id: 2,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!, always has amazing content \n I love this movie'
    },
    {
        id: 3,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: '&^%$#@!'
    }
]

