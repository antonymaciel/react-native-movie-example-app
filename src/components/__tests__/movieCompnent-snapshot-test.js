import React from 'react';
import renderer from 'react-test-renderer';
import { movie } from '../../../config/jest/mockData';
import MovieComponent from '../movieComponent';


it('renders a movieComponent using Snapshots', () => {
  expect(renderer.create(
    <MovieComponent
        year={movie.year}
        imageUrl={movie.imageUrl}
        title={movie.title}
        vote_average ={movie.vote_average}
    />
  )).toMatchSnapshot();
});

