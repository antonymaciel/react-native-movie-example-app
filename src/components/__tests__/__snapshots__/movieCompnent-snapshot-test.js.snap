// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a movieComponent using Snapshots 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginLeft": 10,
      "marginTop": 10,
      "opacity": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg",
      }
    }
    style={
      Object {
        "backgroundColor": "black",
        "borderRadius": 15,
        "height": 620.5,
        "width": 360,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "orange",
        "borderRadius": 100,
        "marginRight": 25,
        "marginTop": 25,
        "padding": 7,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontWeight": "bold",
        }
      }
    >
      9.1
    </Text>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "marginBottom": 20,
        "marginLeft": 20,
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      2019
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 17,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        }
      }
    >
      Spiderman
    </Text>
  </View>
</View>
`;
